oxapy/__init__.py

```
@@ -1,8 +1,19 @@
 from .oxapy import *
 
+import os
 import mimetypes
 
 
+def secure_join(base: str, user_path: str) -> str:
+    base = os.path.abspath(base)
+    target = os.path.abspath(os.path.join(base, user_path))
+
+    if not target.startswith(base + os.sep):
+        raise exceptions.ForbiddenError("Access denied")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)